src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -65,6 +70,28 @@ public DynamicallyAccessedMemberTypes GetFieldAnnotation (FieldDefinition field)
 			return DynamicallyAccessedMemberTypes.None;
 		}
 
+		public DynamicallyAccessedMemberTypes GetGenericParameterAnnotation (GenericParameter genericParameter)
+		{
+			TypeDefinition declaringType = genericParameter.DeclaringType?.Resolve ();
```

Comment:
What about a case like

```c#
class A<T>
{
    class B
    {
        static void Foo ()
        {
			T t;
        }
    }
}
```

will the generic parameter point to the right one with the annotation ?

Reply:
Yes, the C# compiler propagates the attributes on `A`'s `T` to the `B`'s `T`.

We might have reporting that doesn't make sense to end users though (because it will refer to `B`'s generic parameter). I would probably leave fixing that to user feedback - it requires hardcoding C# compiler's details that I'm not sure map to e.g. F#.

Reply:
I'll add a test case for this - it's not great but it's OKish. Since we report the full type name and the actual name of the generic parameter like this ``Generic parameter 'T' from 'A`1/B'`` I think it's acceptable.